examples/http/server.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+#include "opentelemetry/ext/http/server/http_server.h"
+#include<string>
+#include<atomic>
+
+
+namespace {
+
+class HttpServer : public HTTP_SERVER_NS::HttpRequestCallback
+{
+
+protected:
+    HTTP_SERVER_NS::HttpServer server_;
+    std::string server_url_ ;
+    uint16_t port_ ;
+    std::atomic<bool> is_running_{false};
+
+public:
+
+    HttpServer(std::string server_name = "test_server",uint16_t port = 8800): port_(port){
+        server_.setServerName(server_name);
+        server_.setKeepalive(false);
+    }
+
+    void AddHandler(std::string path, HTTP_SERVER_NS::HttpRequestCallback *request_handler){
+        server_.addHandler(path, *request_handler);
+    }
+
+    void Start() {
+        if (!is_running_) {
+            server_.addListeningPort(port_);
```

Comment:
It's probably not in the scope of this PR but shouldn't we be more careful about error result for this operation? Like port already in use or you don't have permissions for listening below 1024.

Reply:
Correct. The original purpose of this http server library was only for testing purpose, and so error handling was not there. We can revisit this if it's used at more places.